libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -104,111 +99,202 @@ - (void)dealloc {
 }
 
 - (void)setOauthCompletionUrl:(NSString *)newRedirectUri
 {
-    NSUserDefaults *defs = [NSUserDefaults msdkUserDefaults];
-    [defs setObject:newRedirectUri forKey:kOAuthRedirectUriKey];
-    [defs synchronize];
+    [SFSDKAuthPreferences sharedPreferences].oauthCompletionUrl = newRedirectUri;
 }
 
 - (NSString *)oauthClientId
 {
-    NSUserDefaults *defs = [NSUserDefaults msdkUserDefaults];
-    NSString *clientId = [defs objectForKey:kOAuthClientIdKey];
-    return clientId;
+    return [SFSDKAuthPreferences sharedPreferences].oauthClientId;
 }
 
 - (void)setOauthClientId:(NSString *)newClientId
 {
-    NSUserDefaults *defs = [NSUserDefaults msdkUserDefaults];
-    [defs setObject:newClientId forKey:kOAuthClientIdKey];
-    [defs synchronize];
+   [SFSDKAuthPreferences sharedPreferences].oauthClientId = newClientId;
+}
+
+#pragma  mark - login & logout
+
+- (BOOL)handleAdvancedAuthenticationResponse:(NSURL *)appUrlResponse {
+    [SFSDKCoreLogger i:[self class] format:@"handleAdvancedAuthenticationResponse %@",[appUrlResponse description]];
+    
+    //should return the shared instance for advanced auth
+    SFSDKOAuthClient *client = [SFSDKOAuthClient nativeBrowserAuthInstance:nil];
+    return [client handleURLAuthenticationResponse:appUrlResponse];
+}
+
+- (BOOL)loginWithCompletion:(SFUserAccountManagerSuccessCallbackBlock)completionBlock failure:(SFUserAccountManagerFailureCallbackBlock)failureBlock {
+    SFOAuthCredentials *clientCredentials = [self retrieveClientCredentials];
+    return [self refreshCredentials:clientCredentials completion:completionBlock failure:failureBlock];
+}
+
+- (BOOL)refreshCredentials:(SFOAuthCredentials *)credentials completion:(SFUserAccountManagerSuccessCallbackBlock)completionBlock failure:(SFUserAccountManagerFailureCallbackBlock)failureBlock {
+
+    SFSDKOAuthClient *client = [SFSDKOAuthClient clientWithCredentials:credentials configBlock:^(SFSDKOAuthClientConfig *config) {
+        config.advancedAuthConfiguration = self.advancedAuthConfiguration;
+        config.delegate = self;
+        config.webViewDelegate = self;
+        config.safariViewDelegate = self;
+        config.successCallbackBlock = completionBlock;
+        config.failureCallbackBlock = failureBlock;
+    }];
+    
+    return [client refreshCredentials];
+}
+
+- (BOOL)loginWithJwtToken:(NSString *)jwtToken completion:(SFUserAccountManagerSuccessCallbackBlock)completionBlock failure:(SFUserAccountManagerFailureCallbackBlock)failureBlock {
+    NSAssert(jwtToken.length > 0, @"JWT token value required.");
+    SFOAuthCredentials *credentials = [self retrieveClientCredentials];
+    credentials.jwt = jwtToken;
+    return [self refreshCredentials:credentials
+                                  completion:completionBlock
+                                     failure:failureBlock];
+}
+
+- (void)logout {
+    [self logoutUser:[SFUserAccountManager sharedInstance].currentUser];
+}
+
+- (void)logoutUser:(SFUserAccount *)user {
```

Comment:
It looks like the guardrails on multiple logout calls were removed from this porting.  See [this PR](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/1989).  This will likely result in regressions when these new interfaces are consumed.